### PR TITLE
Update .net SDK based on latest official swagger spec for sql 2017-03-01-preview

### PR DESCRIPTION
**Pull request by jiawang003**
## Description

1) Update .net SDK based on latest official swagger spec for Sql 2017-03-01-preview, here's the link for that committed PR:
https://github.com/Azure/azure-rest-api-specs/commit/b3f4823378bab4812792ad0b90da6e8a8385a2f1

2) include newly added testcase TestCancelDatabaseOperation for list and cancel database operation API, and new test passed

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
If you are planning ot publish, you will need to bump up the version for your nuget package.

**Comment by shahabhijeet** [comment]
You can use this instead of checking for record mode
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SdkCommon/TestFramework/ClientRuntime.Azure.TestFramework/Utilities/TestUtilities.cs#L84

**Comment by jiawang003** [comment]
updated to use TestUtilities.Wait instead of checking for record mode

**Comment by jiawang003** [comment]
resolved all review comments so far

**Comment by jaredmoo** [comment]
@shahabhijeet the last published version was 1.6 ( https://www.nuget.org/packages/Microsoft.Azure.Management.Sql ) so 1.7 in assemblyversion is already correct.

File: src/SDKs/SqlManagement/Sql.Tests/OperationsScenarioTests.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@ public void TestListOperations()
             {
                 var sqlClient = context.GetClient<SqlManagementClient>();
 
+                /*
                 OperationListResult result = sqlClient.Operations.List();
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/OperationsScenarioTests.cs]: change OperationListResult to IPage<Operation> and uncomment

**Review by jaredmoo — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**